### ISSUE TITLE
taurusLabel.setModelIndex() not working

### ISSUE THREAD

**Issue by CorF4** [taurus-org/taurus#648]
Hello,

in my tango device server I am using some spectrum attributes, some are boolean, some are double or strings. I can display the boolean spectrum attributes  via taurusLed and setting the modelIndex to the corresponding index of the attribute. This works fine. However, I also want to display the string and double spectrum attributes. Using the same method does not work here. Instead of the single entry corresponding to modelIndex the whole array is displayed (if the taurusLabel is large enough). I am wondering if there is a bug with taurusLabel and spectrum attributes (I tested the boolean version with taurusLabel and it didn't work, but I am not 100% sure it is down to taurusLabel or non-booleans). I wanted to ask here first if you can help with that before I need to transform all my spectrum attributes to scalar attributes.

Thanks a lot!

**Comment by cpascual** [comment]
Hi, I reproduced the behaviour that you describe with this simple snippet:

```python
from taurus.qt.qtgui.application import TaurusApplication
from taurus.qt.qtgui.display import TaurusLabel

if __name__ == '__main__':
    import sys
    app = TaurusApplication()

    w = TaurusLabel()

    model = 'eval:10*arange(3)'
    w.setModel(model)
    w.setModelIndex(2)

    w.show()
    sys.exit(app.exec_()
```

The output is a label whose text is `[0 10 20]` instead of the expected `20`.

I think this is a bug in TaurusLabel (it used to work at some time, but it is not a commonly used feature and we did not realize the regression). 

We should fix it and also add unit tests to prevent regressions (but I cannot guarantee to devote time to it before the next release in January). If someone else wants to help...

**Comment by CorF4** [comment]
Hi, thanks for the confirmation!
Indeed, I see that it works in taurus-3.6.0.

**Comment by PhilLAL** [comment]
Hi,
reading the doc (http://www.taurus-scada.org/en/latest/devel/api/taurus/qt/qtgui/display/_TaurusLabel.html#taurus.qt.qtgui.display.TaurusLabel.modelIndex) I wonder what is the expected behaviour for modelIndex?

**Comment by cpascual** [comment]
It is for array attrs. Showing only one item instead of a whole array.

**Comment by cpascual** [comment]
It is for array attrs. Showing only one item instead of a whole array.

**Comment by PhilLAL** [comment]
Ok, thank you for the explanation. 
I suggest completing the doc like this for the newcomer: «This property holds the index inside the model value that should be displayed when the model value is an array attribute.»

**Comment by mkomel** [comment]
Fix proposed by PR #681

**Comment by cpascual** [comment]
closed via #681
